crates/wasmtime/src/component/mod.rs

```diff
@@ -9,13 +9,17 @@ mod instance;
 mod linker;
 mod matching;
 mod store;
+mod types;
```

Comment:
GIven how many types are in this module and how `Type` isn't the only export I think it may actually make sense to have this be a `pub mod types` and don't reexport `Type` from inside. Otherwise external users can't name types like `Field` and such.